### PR TITLE
Check if target is managed by VC

### LINKED ISSUE #1998: Running vic-machine ls on a host that is managed by VC returns an empty list
When you run `vic-machine ls` and specify an ESXi host in the `target ` option, if that ESXi host is managed by vCenter Server, `vic-machine ls` returns an empty list. This happens even if there are VCHs running on that ESXi host in VC. This is potentially misleading.

If possible, `vic-machine ls` should return an error stating that the target host is managed by VC XXX, as is the case when you try to run `vic-machine create` on an ESXi host that is managed by VC.

### PR DESCRIPTION
**Pull request by emlin**
Fixes #1998

### CODE REVIEW

**Comment by hmahmood** [comment]
lgtm

File: lib/install/validate/validator.go [hunk 1]
```diff
@@ -280,19 +289,40 @@ func (v *Validator) target(ctx context.Context, input *data.Data, conf *config.V
...
+
+	var mh mo.HostSystem
+
+	if err = host.Properties(ctx, host.Reference(), []string{"summary.managementServerIp"}, &mh); err != nil {
+		v.NoteIssue(fmt.Errorf("Failed to get host properties: %s", err))
+		return
+	}
+
+	if ip := mh.Summary.ManagementServerIp; ip != "" {
+		v.NoteIssue(fmt.Errorf("Target is managed by vCenter server %q, please change --target to vCenter or standalone ESXi address", ip))
```
Comment (andrewtchin) [inline lib/install/validate/validator.go]: maybe `Target is managed by vCenter server %q, please change --target to the vCenter address or select a standalone ESXi`

**Comment by andrewtchin** [comment]
lgtm

File: lib/install/validate/validator.go (hunk 1 above)
Comment (emlin) [inline lib/install/validate/validator.go]: nice, thanks

**Comment by emlin** [comment]
♻️

**Pull request merged**